### PR TITLE
fix(TypeScript): interfaces that accepted `Scheduler` now accept `ISc…

### PR DESCRIPTION
**Pull request by blesh**
…heduler` interface

this should be a non-breaking change. It just makes sense to accept a minimal interface here rather than a full-on class

The full on class was giving me difficulties in another TypeScript project where I wanted to mock a Scheduler.

### CODE REVIEW

File: src/Scheduler.ts [hunk 1]
```diff
@@ -17,7 +21,7 @@ import { Subscription } from './Subscription';
  *
  * @class Scheduler
  */
-export class Scheduler {
+export class Scheduler implements IScheduler {
 
   public static now: () => number = Date.now ? Date.now : () => +new Date();
```
Comment (kwonoj) [inline src/Scheduler.ts]: if `now` is defined in interface, shouldn't this be instance method as well to allow override?

Comment (blesh) [inline src/Scheduler.ts]: Interesting. I don't think this should be static at all. Not sure how that crept in there.

Comment (trxcllnt) [inline src/Scheduler.ts]: @blesh from what I recall, it's static to mirror the [Rx4 Scheduler API](https://github.com/Reactive-Extensions/RxJS/blob/master/src/core/concurrency/scheduler.js#L45)

Comment (blesh) [inline src/Scheduler.ts]: What was the purpose there? Not that it pertains to this PR

Comment (blesh) [inline src/Scheduler.ts]: What was the purpose there? Not that it pertains to this PR

Comment (trxcllnt) [inline src/Scheduler.ts]: I believe schedulers were at one point static classes in Rx.NET, but could be misremembering.

Comment (kwonoj) [inline src/Scheduler.ts]: one question for this is, if this becomes instance method will this be breaking change for next major semver?

Comment (trxcllnt) [inline src/Scheduler.ts]: It is already both static and instance method (assigned via the constructor)

Comment (kwonoj) [inline src/Scheduler.ts]: 😮 oh, thanks for clarification.

Comment (blesh) [inline src/Scheduler.ts]: If anything, we might deprecate and remove the static version at some point.

**Pull request merged**